api/api.go

```diff
@@ -327,3 +328,42 @@ func GetRepoContributors(url string) []Contributor {
 
 	return contributors
 }
+
+func getPackage(npmUrl string) (packageName string) {
+
+	i := strings.Index(npmUrl, "package")
+	return npmUrl[i+len("package")+1 : len(npmUrl)]
+}
+
+func NPMtoGithubUrl(npmUrl string) (githubUrl string) {
+	npmPackage := getPackage(npmUrl)
+	NPMendpoint := "https://registry.npmjs.org/" + npmPackage
+
+	res, err := http.Get(NPMendpoint)
+
+	if err != nil {
+		fmt.Printf("The HTTP request failed with error %s\n", err)
+	}
+
+	data, err := ioutil.ReadAll(res.Body)
+	all_data := string(data)
+
+	i := strings.Index(all_data, "git+https://github.com/")
+
+	// 60 is an arbitrary number to (surely) capture the entire url name
+	// done so that parsing is not done over entire string of all data 
+	githubUrl = all_data[i : i+60]
+	escapeIndex := 0
+	j := 0
+
+	for j = range githubUrl {
+		if string(githubUrl[j]) == "/" {
+			escapeIndex += 1
+		}
+		if escapeIndex == 4 && string(githubUrl[j]) == "." {
+			break
```

Comment:
what is going on here i do not understand

Reply:
like why do we need escapeindex. why are we parsing the url? what is with the rule of 4 going on. I am lost

Reply:
This will be removed but it was to iterate through the github url extracted plus some characters until the fourth forward slash was found. After this is found, one can just add the length of the package name to get the full url.